src/components/VideoPlayer/BaseVideoPlayer.js

```diff
@@ -107,34 +120,33 @@ function BaseVideoPlayer({
             }
 
             preventPausingWhenExitingFullscreen(isVideoPlaying);
-            setIsPlaying(isVideoPlaying);
+            updateIsPlaying(isVideoPlaying);
             setIsLoading(!e.isLoaded || Number.isNaN(e.durationMillis)); // when video is ready to display duration is not NaN
             setIsBuffering(e.isBuffering || false);
             setDuration(currentDuration);
             setPosition(currentPositon);
 
             onPlaybackStatusUpdate(e);
         },
-        [onPlaybackStatusUpdate, preventPausingWhenExitingFullscreen, videoDuration],
+        [onPlaybackStatusUpdate, preventPausingWhenExitingFullscreen, updateIsPlaying, videoDuration],
     );
 
     const handleFullscreenUpdate = useCallback(
         (e) => {
             onFullscreenUpdate(e);
-
             // fix for iOS native and mWeb: when switching to fullscreen and then exiting
             // the fullscreen mode while playing, the video pauses
-            if (!isPlaying || e.fullscreenUpdate !== VideoFullscreenUpdate.PLAYER_DID_DISMISS) {
-                return;
-            }
-
-            if (isMobileSafari) {
-                pauseVideo();
+            if (e.fullscreenUpdate === VideoFullscreenUpdate.PLAYER_DID_DISMISS) {
+                isFullscreen.current = false;
+                // we  have to use ref to check if the video is playing, because in other case state varaible change will dismiss fullscreen mode
```

Comment:
`varaible` type

Reply:
this is probably a good comment for people that know this codebase, but for me it doesn't help me that much :/ I still don't know why the state would dismiss the video.

Reply:
Me either 😅 After some research I think that state change dismiss the fullscreen mode because the `useCallback` reference is changing on fly when videos is using it

Reply:
Probably it's a bug in `expo-av`